connection/grpctest/server.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpctest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/apigee/registry/server/registry"
+	"google.golang.org/grpc"
+)
+
+// NewIfNoAddress will create a RegistryServer served by a
+// basic grpc.Server if env var APG_REGISTRY_ADDRESS is
+// not set, see NewServer() for details. If APG_REGISTRY_ADDRESS
+// is set, returns nil as client will connect to that remote address.
+// Example:
+// func TestXXX(t *testing.T) {
+// 	 l, err := grpctest.NewIfNoAddress(registry.Config{})
+// 	 if err != nil {
+// 		t.Fatal(err)
+// 	 }
+// 	 defer l.Close()
+//   ... run test here ...
+// }
+func NewIfNoAddress(rc registry.Config) (*Server, error) {
+	addr := os.Getenv("APG_REGISTRY_ADDRESS")
+	if addr != "" {
+		log.Printf("Client will use remote database at: %s", addr)
+		return nil, nil
+	}
+	log.Println("Client will use embedded SQLite3 database")
```

Comment:
Is there a way (or is there no way) to configure the automatically-generated server to use a Postgres database?

Reply:
I want to be able to detect (and rule-out) database-specific problems that might appear in the cmd tests.

Most database-specific problems should be found by the core server tests (in server/registry), and these can be run locally with postgres, which is great. If we assume that the core tests are complete, then we wouldn't need to run the cmd tests against a postgres database. But to me, that assumption takes an unnecessary risk for the arguable convenience of not having to start a test server.

Reply:
Aside: I'm just going to drop this link here in case we decide to run unit tests with an embedded postgres: https://github.com/fergusstrange/embedded-postgres.

Reply:
I just created #624 to evaluate embedded postgres for unit and CI testing.

Summarizing our face-to-face discussion, we agreed to move forward with this PR with a new near-term goal to replace the use of environment variables to configure the registry client with configuration that can be read from a file. This file can be managed by the registry tool (allowing commands like `registry auth login` and `registry config set location`) and could point to arbitrary servers with arbitrary database backends. We think this is useful for allowing `registry config set project` and `registry config set location`, which could allow shorter relative resource names to be provided to other `registry` subcommands like `registry list`.

Most relevant to this PR, by removing the dependence on global values (environment variables) to configure tests, that reduces the uncertainty about what is happening that is expressed in all the comments like this one:
```
// TestMain will set up a local RegistryServer and grpc.Server for all
// tests in this package if APG_REGISTRY_ADDRESS env var is not set
// for the client.
```

tl;dr for now we're moving to a slightly more complex state that fixes a short-term problem and will make future improvements (issue TBD) to improve overall simplicity and clarity.